templates/skeleton/app/routes/cart.$lines.tsx

```diff
@@ -18,51 +18,61 @@ import {redirect, type LoaderFunctionArgs} from '@shopify/remix-oxygen';
  *
  * ```
  */
-export async function loader({request, context, params}: LoaderFunctionArgs) {
-  const {cart} = context;
-  const {lines} = params;
-  if (!lines) return redirect('/cart');
-  const linesMap = lines.split(',').map((line) => {
-    const lineDetails = line.split(':');
-    const variantId = lineDetails[0];
-    const quantity = parseInt(lineDetails[1], 10);
+export const loader = defineLoader(
+  async ({request, context, params, response}) => {
+    const {cart} = context;
+    const {lines} = params;
+    if (!lines)
+      throw new Response(null, {
+        status: 302,
+        headers: {
+          Location: '/cart',
+        },
+      });
+    const linesMap = lines.split(',').map((line) => {
+      const lineDetails = line.split(':');
+      const variantId = lineDetails[0];
+      const quantity = parseInt(lineDetails[1], 10);
 
-    return {
-      merchandiseId: `gid://shopify/ProductVariant/${variantId}`,
-      quantity,
-    };
-  });
+      return {
+        merchandiseId: `gid://shopify/ProductVariant/${variantId}`,
+        quantity,
+      };
+    });
 
-  const url = new URL(request.url);
-  const searchParams = new URLSearchParams(url.search);
+    const url = new URL(request.url);
+    const searchParams = new URLSearchParams(url.search);
 
-  const discount = searchParams.get('discount');
-  const discountArray = discount ? [discount] : [];
+    const discount = searchParams.get('discount');
+    const discountArray = discount ? [discount] : [];
 
-  // create a cart
-  const result = await cart.create({
-    lines: linesMap,
-    discountCodes: discountArray,
-  });
+    // create a cart
+    const result = await cart.create({
+      lines: linesMap,
+      discountCodes: discountArray,
+    });
 
-  const cartResult = result.cart;
+    const cartResult = result.cart;
 
-  if (result.errors?.length || !cartResult) {
-    throw new Response('Link may be expired. Try checking the URL.', {
-      status: 410,
-    });
-  }
+    if (result.errors?.length || !cartResult) {
+      throw new Response('Link may be expired. Try checking the URL.', {
+        status: 410,
+      });
+    }
 
-  // Update cart id in cookie
-  const headers = cart.setCartId(cartResult.id);
+    // Update cart id in cookie
+    cart.setCartId(cartResult.id, response);
```

Comment:
Maybe response can be pass in from `cart.create` and our utility can take care of the error redirect + setCardId?

Need some help from Remix team, but if response can be passed in from `createCartHandler` that would be really ideal.